src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -202,33 +202,32 @@ void checkAndWarnIgnored(@Nullable Object propertyToIgnore, String propertyName)
   }
 
   /**
-   * This method tries to guess the runtime based on the appengine-web.xml of all services that are
-   * expected to run.
+   * This method determines if the runtime based on the appengine-web.xml of all services that are
+   * expected to run can run without sandbox restrictions.
    *
    * @param services a list of app engine standard service directories
-   * @return {@code false} if only java7 modules are found or {@code true} is at least one java8
-   *     module is found (i.e. pure java8 or mixed java7/java8)
+   * @return {@code true} if there no need to enforce the sandbox restrictions.
    */
   @VisibleForTesting
-  boolean isJava8(List<File> services) throws AppEngineException {
-    boolean java8Detected = false;
+  boolean isWhiteListEnforced(List<File> services) throws AppEngineException {
```

Comment:
`isWhiteListEnforced` -> `isSandboxEnforced` ?

Reply:
Because of the renaming,

```diff
-    String gaeRuntime = getGaeRuntimeJava(isJava8);
+    String gaeRuntime = getGaeRuntimeJava(!isWhiteListEnforced);
```

I think we should update the Javadoc, rename the argument of `getGaeRuntimeJava()`, and maybe change the contract of it too.

```java
  /**
   * Gets the App Engine runtime ID for Java runtimes.
   *
   * @param isJava8 if {@code true}, use Java 8; otherwise, use Java 7
   * @return "java8" if {@code isJava8} is true; otherwise, returns "java7"
   */
  @VisibleForTesting
  static String getGaeRuntimeJava(boolean isJava8) {
    return isJava8 ? "java8" : "java7";
  }
```